cudamapper/CMakeLists.txt

```diff
@@ -22,86 +22,51 @@ else()
     set(CUDA_NVCC_FLAGS "${CUDA_NVCC_FLAGS} -lineinfo --expt-extended-lambda -use_fast_math -Xcompiler -Wall,-Wno-pedantic -std=c++14")
 endif()
 
-cuda_add_library(minimizer
-        src/minimizer.cu)
-target_include_directories(minimizer PUBLIC include ${CUB_DIR})
-target_link_libraries(minimizer cgabase cgalogging pthread)
-target_compile_options(minimizer PRIVATE -Werror)
-
-cuda_add_library(index_gpu
+cuda_add_library(cudamapper
+        src/cudamapper.cpp
+        src/index_descriptor.cpp
         src/index.cu
+        src/index_cache.cu
         src/index_gpu.cu
         src/index_host_copy.cu
-        src/minimizer.cu)
-target_include_directories(index_gpu PUBLIC include ${CUB_DIR})
-target_link_libraries(index_gpu cgabase cgaio cgalogging minimizer pthread)
-target_compile_options(index_gpu PRIVATE -Werror)
-
-cuda_add_library(matcher_gpu
-        src/matcher_gpu.cu)
-target_include_directories(matcher_gpu PUBLIC include ${CUB_DIR})
-target_link_libraries(matcher_gpu cgabase cgaio cgalogging)
-target_compile_options(matcher_gpu PRIVATE -Werror)
-
-add_library(cudamapper_utils
-        src/cudamapper_utils.cpp)
-target_include_directories(cudamapper_utils PUBLIC include)
-target_link_libraries(cudamapper_utils cgabase)
+        src/minimizer.cu
+        src/matcher.cu
+        src/matcher_gpu.cu
+        src/cudamapper_utils.cpp
+        src/overlapper.cpp
+        src/overlapper_triggered.cu)
 
-cuda_add_library(overlapper_triggerred
-        src/overlapper_triggered.cu
-        src/overlapper.cpp)
-target_include_directories(overlapper_triggerred PUBLIC include ${CUB_DIR})
-target_link_libraries(overlapper_triggerred cgabase cgaio cgalogging cudaaligner)
-target_compile_options(overlapper_triggerred PRIVATE -Werror)
+target_include_directories(cudamapper
+    PUBLIC
+        $<INSTALL_INTERFACE:include>
+        $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+)
+target_link_libraries(cudamapper cgabase cgaio cgalogging cub pthread)
```

Comment:
BTW, I realized we don't have to link `pthread` here or in `common/logging`. Can you please remove them as part of this PR?

Reply:
I can see how `logging` does not need it, but `cudamapper`? We use `std::thread` in `cudamapper`, doesn't that require `pthread` anymore?

Reply:
I would assume now it links it automatically as it is part of standard...

Reply:
I remember that it was required in the past, even though it was part of the standard library. Anyhow, that may have changed. removed.